src/App.tsx

```diff
@@ -1,7 +1,31 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    message: 'Nothing was pressed yet',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.handlerKeyUp);
+  }
+
+  componentDidUpdate() {
+    document.removeEventListener('keyup', this.handlerKeyUp);
```

Comment:
You should remove the listener on the unmount, not on the update. Now the handlerKeyUp works only after the first click. But it should be called on every keyboard press.